parcels/field.py

```diff
@@ -288,41 +263,38 @@ def from_netcdf(cls, name, dimensions, filenames, indices={},
-            time_i = range(np.where(self.grid.time >= timerange[0])[0][0], np.where(self.grid.time <= timerange[1])[0][-1]+1)
-            time = self.grid.time[time_i]
-        if lonrange is None:
-            lon_i = range(len(self.grid.lon))
-            lon = self.grid.lon
-        else:
-            lon_i = range(np.where(self.grid.lon >= lonrange[0])[0][0], np.where(self.grid.lon <= lonrange[1])[0][-1]+1)
-            lon = self.grid.lon[lon_i]
-        if latrange is None:
-            lat_i = range(len(self.grid.lat))
-            lat = self.grid.lat
-        else:
-            lat_i = range(np.where(self.grid.lat >= latrange[0])[0][0], np.where(self.grid.lat <= latrange[1])[0][-1]+1)
-            lat = self.grid.lat[lat_i]
-
-        dVdx = np.zeros(shape=(time.size, lat.size, lon.size), dtype=np.float32)
-        dVdy = np.zeros(shape=(time.size, lat.size, lon.size), dtype=np.float32)
-        for t in np.nditer(np.int32(time_i)):
-            grad = CentralDifferences(np.transpose(self.data[t, :, :][np.ix_(lat_i, lon_i)]), lat, lon)
-            dVdx[t, :, :] = np.array(np.transpose(grad[0]))
-            dVdy[t, :, :] = np.array(np.transpose(grad[1]))
-
-        return([Field(name + '_dx', dVdx, lon=lon, lat=lat, depth=self.grid.depth, time=time,
-                      interp_method=self.interp_method, allow_time_extrapolation=self.allow_time_extrapolation),
-                Field(name + '_dy', dVdy, lon=lon, lat=lat, depth=self.grid.depth, time=time,
-                      interp_method=self.interp_method, allow_time_extrapolation=self.allow_time_extrapolation)])
+    def cell_edge_sizes(self):
+        """Method to calculate cell sizes based on numpy.gradient method
+                Currently only works for Rectilinear Grids"""
+        if self.grid.gtype not in (GridCode.RectilinearZGrid, GridCode.RectilinearSGrid):
+            logger.error('Field.cell_edge_sizes() is only implemented for Rectilinear Grids')
+            exit(-1)
+        dy_grid = np.zeros((self.grid.lon.size, self.grid.lat.size), dtype=np.float32)
+        dx_grid = np.zeros((self.grid.lon.size, self.grid.lat.size), dtype=np.float32)
+
+        x_conv = GeographicPolar() if self.grid.mesh is 'spherical' else UnitConverter()
+        y_conv = Geographic() if self.grid.mesh is 'spherical' else UnitConverter()
+        for y, (lat, dy) in enumerate(zip(self.grid.lat, np.gradient(self.grid.lat))):
+            for x, (lon, dx) in enumerate(zip(self.grid.lon, np.gradient(self.grid.lon))):
+                dx_grid[x, y] = x_conv.to_source(dx, lon, lat, self.grid.depth[0])
+                dy_grid[x, y] = y_conv.to_source(dy, lon, lat, self.grid.depth[0])
+        return dx_grid, dy_grid
+
+    def cell_areas(self):
+        """Method to calculate cell sizes based on cell_edge_sizes
+                Currently only works for Rectilinear Grids"""
+        dx_mesh, dy_mesh = self.cell_edge_sizes()
+        return dx_mesh * dy_mesh
+
+    def gradient(self):
```

Comment:
You put in the PR comment that gradient is only available for rectilinear grids, but I would also put it here in the code.